### ISSUE TITLE
Move content revisions into a module so that it can be disabled easily

### ISSUE THREAD

**Issue by jenlampton** [backdrop/backdrop-issues#284]
It's always bothered me that sites that don't need node revisions have no way of turning them off.  Can we move this functionality into a separate module, so that it can be disabled by default?  (It's not a 80% use-case)

**Comment by BWPanda** [comment]
Is this issue still needed? Not sure if this was an option when this issue was submitted, but there's an setting to 'Show option to create new revisions' in BD now, so does that solve the issue of disabling revisions, or do we still want to discuss moving them to their own module?

If yes, I'm in favour of it, as I never use them.

**Comment by jenlampton** [comment]
I'm in favor of it as well, as I also never use them. For me, the work of disabling revisions on every content type is still too much work. I'd prefer to have single off switch for my whole site.

**Comment by docwilmot** [comment]
Would it be simpler to just have a switch at `admin/config/content` that turns revisions off? Dont know how much work it would be to disentangle the concept of revisions from nodes/entities.

**Comment by jenlampton** [comment]
I don't think we need to disentangle the concept of revisions from nodes/entities, we just need the module to be the "switch". Then we can move all the revision specific code into the module, and the core module can be a model for how contrib can deal with revisions on other entities, for example. The entity API still needs to deal with revisions or contrib wouldn't have the ability to use them.

**Comment by olafgrabienski** [comment]
> For me, the work of disabling revisions on every content type is still too much work.

What about disabling the `Show option to create new revisions` checkbox by default?

**Comment by jenlampton** [comment]
I think that would be too much work for those that do want revisions to
turn it on for every content type. It should be a module, to be consistent
with everything else.

**Comment by olafgrabienski** [comment]
@jenlampton I see! Personally, I don't find it much work to enable or disable revisions via the content type settings. But if it makes also sense from an architecture point of view to put revisions in a module, that may be a good decision.

I'm however still a bit skeptical about the proposal because the option to use revisions is better visible when revisions stay in the settings, especially for users new to Backdrop. (I don't think that many users scan the *Modules* page to find such things.) What do you (and others) think about it?

**Comment by BWPanda** [comment]
Since revisions are enabled by default on new content types currently, making Revisions into a module would mean having that module also enabled by default. People like Jen and I could then easily disabled/uninstall revisions, and everyone else still sees and uses them as normal.

**Comment by jenlampton** [comment]
I think we should start by leaving them enabled in core, but in a module to
easily turn off site-wide.
